clientconn.go

```diff
@@ -196,9 +196,9 @@ func WithTimeout(d time.Duration) DialOption {
 }
 
 // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
-func WithDialer(f func(addr string, timeout time.Duration) (net.Conn, error)) DialOption {
+func WithDialer(d net.Dialer) DialOption {
```

Comment:
It's not a real leak - it's that the networking stack is not being
cancelled and blocking, looking like a leak and causing our tests to flake.
On Jul 6, 2016 22:11, "Qi Zhao" <notifications@github.com> wrote:

> In clientconn.go
> <https://github.com/grpc/grpc-go/pull/751#discussion_r69842457>:
>
> > @@ -196,9 +196,9 @@ func WithTimeout(d time.Duration) DialOption {
> >  }
> >
> >  // WithDialer returns a DialOption that specifies a function to use for dialing network addresses.
> > -func WithDialer(f func(addr string, timeout time.Duration) (net.Conn, error)) DialOption {
> > +func WithDialer(d net.Dialer) DialOption {
>
> I might have thought about something else you did not intend to tackle
> here. To be clear, what goroutine is leaked? Is it a real leak or just the
> stress is too intensive to get that goroutine eliminated in time?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/grpc/grpc-go/pull/751/files/497877d020a44cfed1f3b09ceb7e5e4f61ae32a2#r69842457>,
> or mute the thread
> <https://github.com/notifications/unsubscribe/ABdsPNG1iuCcqlTxrdzD71UK2pvDY2tPks5qTGA3gaJpZM4JFbEQ>
> .
>